homeassistant/components/sensor/fints.py

```diff
@@ -0,0 +1,288 @@
+            if not account_name:
+                account_name = '{} - {}'.format(
+                    fints_name, account.accountnumber)
+            accounts += [FinTsHoldingsAccount(
+                client, account, account_name)]
+            _LOGGER.debug('Creating account %s for bank %s',
+                          account.accountnumber, fints_name)
+        else:
+            _LOGGER.info('skipping account %s for bank %s',
+                         account.accountnumber, fints_name)
+
+    add_devices(accounts, True)
+
+
+class FinTsClient(object):
+    """Wrapper around the FinTS3PinTanClient.
+
+    Use this class as Context Manager to get the FinTS3Client object.
+    """
+
+    def __init__(self, credentials: BankCredentials, name: str):
+        """Constructor for class FinTsClient."""
+        self._credentials = credentials
+        self.name = name
+
+    @property
+    def client(self):
+        """Get the client object.
+
+        Opens a new connection if required. Before using the client
+        acquire the lock!
+        """
+        from fints.client import FinTS3PinTanClient
+        return FinTS3PinTanClient(
+            self._credentials.blz, self._credentials.login,
+            self._credentials.pin, self._credentials.url)
+
+    def detect_accounts(self):
+        """Identify the accounts of the bank."""
+        from fints.dialog import FinTSDialogError
+        balance_accounts = []
+        holdings_accounts = []
+        for account in [a for a in self.client.get_sepa_accounts()]:
+            try:
+                self.client.get_balance(account)
+                balance_accounts += [account]
+            except IndexError:
+                # account is not a balance account.
+                pass
+            except FinTSDialogError:
+                # account is not a balance account.
+                pass
+            try:
+                self.client.get_holdings(account)
+                holdings_accounts += [account]
+            except FinTSDialogError:
+                # account is not a holdings account.
+                pass
+
+        return balance_accounts, holdings_accounts
+
+
+class FinTsAccount(Entity):
+    """Sensor for a FinTS balanc account.
+
+    A balance account contains an amount of money (=balance). The amount may
+    also be negative.
+    """
+
+    def __init__(self, client: FinTsClient, account, name: str) -> None:
+        """Constructor for class FinTsAccount."""
+        self._client = client  # type: FinTsClient
+        self._account = account
+        self._name = name  # type: str
+        self._balance = None   # type: float
+        self._currency = None  # type: str
+
+    @property
+    def should_poll(self) -> bool:
+        """Data needs to be polled from the bank servers."""
+        return True
+
+    def update(self) -> None:
+        """Get the current balance and currency for the account."""
+        bank = self._client.client
```

Comment:
Are you sure you want to get a new client every time you do an update and for every bank account?

Reply:
Well right now there is not benefit in caching the client object. The `fints` library opens a new `requests` session for every call you make. I'm not sure if that is the way it's meant to be, but right now there is not benefit in keeping the `client` object alive. 

But that's  a good point. I added a code comment.